topydo/lib/PrettyPrinterFilter.py

```diff
@@ -173,7 +173,28 @@ def filter(self, p_todo_str, p_todo):
         p_todo_str = self.format
 
         for placeholder, getter in placeholders.items():
-            p_todo_str = re.sub(r'%\[?{}\]?'.format(placeholder), getter(p_todo), p_todo_str)
+            repl = getter(p_todo)
+            pattern = r'%({{.+?}})?\[?({})\]?({{.+?}})?'.format(placeholder)
```

Comment:
As suggested on Gitter, I think this regexp deserves to have named match groups.